solana/spl_initialized_account/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+            owner: bs58::decode(&initialized_account.owner).into_vec().unwrap(),
+        };
+
+        let mut buf = Vec::new();
+        prost::Message::encode(&account_owner, &mut buf).unwrap();
+
+        let any = Any {
+            type_url: "type.googleapis.com/sf.substreams.solana.spl.v1.AccountOwner".to_string(),
+            value: buf,
+        };
+
+        let entry = Entry {
+            key: bs58::decode(account).into_vec().unwrap(),
+            value: Some(any),
+        };
+        // substreams::log::info!("adding key: {}", account);
+        entries.push(entry);
+    }
+
+    Ok(Entries { entries })
+}
+
+/// Iterates over successful transactions in given block and take ownership.
+fn transactions_owned(transactions: solTransactions) -> impl Iterator<Item = ConfirmedTransaction> {
+    transactions.transactions.into_iter().filter_map(|trx| -> Option<ConfirmedTransaction> {
+        if let Some(meta) = &trx.meta {
+            if meta.err.is_none() {
+                // Convert between protobuf types by serializing and deserializing
+                let mut buf = Vec::new();
+                if Message::encode(&trx, &mut buf).is_ok() {
+                    if let Ok(converted) = ConfirmedTransaction::decode(&buf[..]) {
+                        return Some(converted);
+                    }
+                }
+            }
+        }
+        None
+    })
+}
+
+fn process_instruction(output: &mut OutputInstructions, compile_instruction: &InstructionView) {
+    let trx_hash = &bs58::encode(compile_instruction.transaction().hash()).into_string();
+    match compile_instruction.program_id().to_string().as_ref() {
+        SOLANA_TOKEN_PROGRAM_KEG | SOLANA_TOKEN_PROGRAM_ZQB => {
+            match process_token_instruction(output, compile_instruction, compile_instruction.meta()) {
+                Err(err) => {
+                    substreams::log::info!("Skipping unknown token instruction in tx {}: {}", trx_hash, err);
+                }
+                _ => {}
+            }
+        }
+        _ => {
+            process_inner_instruction(compile_instruction, trx_hash, compile_instruction.meta(), output);
+        }
+    }
+}
+
+fn process_inner_instruction(
+    compile_instruction: &InstructionView,
+    trx_hash: &String,
+    meta: &substreams_solana::pb::sf::solana::r#type::v1::TransactionStatusMeta,
+    output: &mut OutputInstructions,
+) {
+    for inner in compile_instruction.inner_instructions() {
+        match inner.program_id().to_string().as_ref() {
+            SOLANA_TOKEN_PROGRAM_KEG | SOLANA_TOKEN_PROGRAM_ZQB => {
+                match process_token_instruction(output, &inner, meta) {
+                    Err(err) => {
+                        substreams::log::info!("Skipping unknown inner token instruction in tx {}: {}", trx_hash, err);
+                    }
+                    _ => {}
+                }
+            }
+            _ => {}
+        }
+    }
+}
```

Comment:
We have a `walk_instructions` in `substreams_solana`, we should probably leveraged it instead of re-inventing again a walk.